include/up-cpp/communication/Subscriber.h

```diff
@@ -43,17 +43,18 @@ struct Subscriber {
 	/// @returns * A unique_ptr to a Subscriber if the callback was
 	///            successfully registered.
 	///          * A UStatus with the appropriate failure code otherwise.
-	[[nodiscard]] utils::Expected<std::unique_ptr<Subscriber>, v1::UStatus>
+	[[nodiscard]] static utils::Expected<std::unique_ptr<Subscriber>,
+	                                     v1::UStatus>
 	subscribe(std::shared_ptr<transport::UTransport> transport,
 	          const v1::UUri& topic,
 	          transport::UTransport::ListenCallback&& callback);
 
-protected:
+public:
 	/// @brief Constructor
 	///
 	/// @param transport Transport this subscriber is connected to.
 	/// @param subscription Handle to the callback registered with UTransport.
-	Subscriber(std::shared_ptr<transport::UTransport> transport,
+	Subscriber(const std::shared_ptr<transport::UTransport>& transport,
```

Comment:
`shared_ptr` should be passed by value

```suggestion
	Subscriber(std::shared_ptr<transport::UTransport> transport,
```

Reply:
Done